code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -326,8 +329,14 @@
 		interact(user)
 		return TRUE
 	else
+		if(user.a_intent == INTENT_HELP)
+			for(var/obj/item/integrated_circuit/input/objscaner/S in assembly_components)
+				if(S.check_then_do_work())
+					S.scan(I,user)
+					return TRUE
```

Comment:
Just make a generic hook for integrated circuits instead of looking for objscanners in particular.

Reply:
Generic hook?where can I see,what is it?